src/content/en/updates/2017/09/visual-viewport-api.md

```diff
@@ -0,0 +1,311 @@
+        viewport, in CSS pixels.
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <code>pageLeft</code>
+      </td>
+      <td>
+        Distance between the left edge of the visual viewport, and the left
+        boundary of the document, in CSS pixels.
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <code>pageTop</code>
+      </td>
+      <td>
+        Distance between the top edge of the visual viewport, and the top
+        boundary of the document, in CSS pixels.
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <code>width</code>
+      </td>
+      <td>
+        Width of the visual viewport in CSS pixels.
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <code>height</code>
+      </td>
+      <td>
+        Height of the visual viewport in CSS pixels.
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <code>scale</code>
+      </td>
+      <td>
+        The scale applied by pinch-zooming. If content is twice the size due to
+        zooming, this would return <code>2</code>. This is not affected by
+        <code>devicePixelRatio</code>.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+There are also a couple of events:
+
+    window.visualViewport.addEventListener('resize', listener);
+
+<table class="responsive">
+  <tbody>
+    <tr>
+      <th colspan=2><code>visualViewport</code> events</th>
+    </tr>
+    <tr>
+      <td>
+        <code>resize</code>
+      </td>
+      <td>
+        Fired when <code>width</code>, <code>height</code>, or
+        <code>scale</code> changes.
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <code>scroll</code>
+      </td>
+      <td>
+        Fired when <code>offsetLeft</code> or <code>offsetTop</code> changes.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Demo
+
+The video at the start of this article was created using `visualViewport`,
+[check it out in Chrome 61+](https://visual-viewport.glitch.me/). It uses
+`visualViewport` to make the mini-map stick to the top-right of the visual
+viewport, and applies an inverse scale so it always appears the same size,
+despite pinch-zooming.
+
+## Gotchas
+
+### Events only fire when the visual viewport changes
+
+It feels like an obvious thing to state, but it caught me out when I first
+played with `visualViewport`.
+
+If the layout viewport resizes but the visual viewport doesn't, you don't get a
+`resize` event. However, it's unusual for the layout viewport to resize without
+the visual viewport also changing width/height.
+
+The real gotcha is scrolling. If scrolling occurs, but the visual viewport
+remains static *relative to the layout viewport*, you don't get a `scroll` event
+on `visualViewport`, and this is really common. During regular document
+scrolling, the visual viewport stays locked to the top-left of the layout
+viewport, so `scroll` *does not fire* on `visualViewport`.
+
+If you're wanting to hear about all changes to the visual viewport, including
```

Comment:
Alternatively, you can use `position: fixed` so that you really only care about changes to the visualViewport itself.

Reply:
True. That's what I'm doing in the demo.